Support `ProjectConfig.dbt_project_path = None` & different paths for Rendering and Execution

MrBones757 opened the pull request:
## Description

This MR finishes the work that was started in #605 to add full support for ProjectConfig.dbt_project_path = None, and implements #568.

Within this PR, several things have been updated:

1 - Added project_path fields to RenderConfig and ExecutionConfig
2 - Simplified the consumption of RenderConfig in the dbtGraph class
3 - added option to configure different dbt executable for Rendering vs Execution.

## Related Issue(s)

Closes: #568

## Breaking Change?

None

## Checklist

- [ ] I have made corresponding changes to the documentation (if required)
- [x] I have added tests that prove my fix is effective or that my feature works

MrBones757 commented:
Docs changes still need to be completed, though all the core logic and baseline tests in its current state has been implemented.

@tatiana could you please have a scan over the changes in this MR and let me know if there are any other changes you'd like to see before i finalize the docs / tests?

tatiana reviewed (commented):
@MrBones757, this looks great, thank you!

I left two minor comments - I'll review them again after the checks are passed.

MrBones757 commented on cosmos/config.py:
```diff
@@ -40,6 +40,13 @@ class RenderConfig:
     dbt_deps: bool = True
     node_converters: dict[DbtResourceType, Callable[..., Any]] | None = None
+    dbt_executable_path: str | Path = get_system_dbt()
+    dbt_project_path: InitVar[str | Path | None] = None
+
+    project_path: Path | None = field(init=False)
```
I was experimenting with a feature of data classes here.
When the user creates an object here, they use dbt_project_path, but dbt_project_path is not an attribute of the class once constructed

dbt_project_path is only used for object generation and is passed directly to the __post_init__

project_path on the other hand, cant be defined using the init, but, it can be accessed as an object property, which, to your point will guarantee it being Path or None, simiar to what i did in ProjectConfig, without requiring a custom constructor.

MrBones757 commented:
I've added some additional tests to captures some more edge cases i found and add additional coverage.
Additionally, I've updated some docs, and docstrings, to reflect changes that have occurred in some Config Classes both within this PR and others.

All unit tests are running fine locally, and all integration tests other than databricks are too (no token for this)
I'll verify these in CI and once thats complete its looking just about complete.

MrBones757 commented:
Since the initial integration tests look good, is there anything else that needs to be done here :)
